src/Workspaces/Core/Portable/Workspace/Solution/DocumentState.cs

```diff
@@ -541,13 +541,11 @@ internal DocumentState UpdateTree(SyntaxNode newRoot, PreservationMode mode)
 
             // determine encoding
             Encoding? encoding;
-            ImmutableDictionary<string, ReportDiagnostic>? treeDiagnosticReportingOptions = null;
 
             if (this.TryGetSyntaxTree(out var priorTree))
             {
                 // this is most likely available since UpdateTree is normally called after modifying the existing tree.
                 encoding = priorTree.Encoding;
-                treeDiagnosticReportingOptions = priorTree.DiagnosticOptions;
```

Comment:
Hmm, is your take that this optimization just isn't necessary anymore? Or is this because we can't fetch the generated state *back* from a tree we have to always do the work below regardless?

Reply:
Yes, the latter. Do you foresee any perf issue? I don't think we have any alternative though, unless compiler exposes this flag as a public API on SyntaxTree.